R/get-datasets.R

```diff
@@ -44,18 +48,28 @@ datasets <- function (x=getAPIRoot()) {
 #' the primary dataset catalog for the user will be used.
 #' @param refresh logical: should the function check the Crunch API for new
 #' datasets? Default is FALSE.
+#' @param shiny logical: launch a shiny gadget to help select the right dataset. The
+#' gadget will return a valid `loadDataset()` call which loads the selected dataset.
 #' @return Character vector of dataset names, each of which would be a valid
 #' input for [loadDataset()]
 #' @export
 listDatasets <- function (kind=c("active", "all", "archived"), project=NULL,
-                          refresh=FALSE) {
-    dscat <- selectDatasetCatalog(kind, project, refresh)
-    return(names(dscat))
+    refresh=FALSE, shiny = FALSE) {
+    if (shiny) {
+        if (all(c("rstudioapi", "shiny", "miniUI") %in% rownames(installed.packages())) &&
+                rstudioapi::isAvailable("0.99.878")) {
+            listDatasetGadget(kind, refresh)
+        } else {
+            halt("Dataset viewer can only be accessed from RStudio versions 0.99.878 or later.")
```

Comment:
We should have some reference to the packages if that's what's causing the error (I would split this into two blocks, one that checks if the packages are there, and the other that checks RStudio version. That way the error message can be tailored to the package or the version problem.